lib/ruby_smb/dcerpc/srvsvc.rb

```diff
@@ -0,0 +1,96 @@
+module RubySMB
+  module Dcerpc
+    module Srvsvc
+
+      OPNUMS = {
+          net_share_enum_all: 15
+      }
+
+      #https://msdn.microsoft.com/en-us/library/cc247293.aspx
+      class NetShareEnumAll < BinData::Record
+
+        endian :little
+
+        uint32 :referent_id, initial_value: 0x00000001
+        uint32 :max_count,    initial_value: -> { (server_unc.do_num_bytes / 2) + 1 }
+        uint32 :offset,       initial_value: 0
+        uint32 :actual_count, initial_value: -> {max_count}
+        string :server_unc,   pad_front: false,
+               initial_value: -> {host.encode('utf-16le')}
+
+        uint16 :padding, initial_value: 0
```

Comment:
I think we should calculate the proper padding to follow the NDR alignement specifications:
- [Alignment of Primitive Types](http://pubs.opengroup.org/onlinepubs/9629399/chap14.htm#tagcjh_19_02_02)
- [Alignment of Constructed Types](http://pubs.opengroup.org/onlinepubs/9629399/chap14.htm#tagcjh_19_03_02)